src/modules-load/modules-load.c

```diff
@@ -52,12 +53,30 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent) {
+static void *worker_thread(void *arg) {
+        int sock_fd = *(int *) arg;
+        char buffer[32];
+        ssize_t bytes_received;
+        _cleanup_(sym_kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        int r = module_setup_context(&ctx);
+        if (r < 0)
+                log_error_errno(r, "Failed to initialize libkmod context: %m");
+
+        while ((bytes_received = recv(sock_fd, buffer, sizeof(buffer), 0)) > 0) {
+                buffer[bytes_received] = '\0';
+                r = module_load_and_warn(ctx, buffer, true);
+        }
+
+        // Clean up and close the socket
+        safe_close(sock_fd);
```

Comment:
no, this is wrong. let the socket be. other trheads might still want to use it. This is the receiving side, and he main thread should just close that when every thread is gone